demos/shadow/shadow_demo_main/core_mqtt_config.h

```diff
@@ -0,0 +1,78 @@
+/*
```

Comment:
Why is this file added as a new file? Wasn't there a mqtt_config.h already?

Reply:
Nope, it wasn't present before in the Shadow demo. It was instead utilizing the mqtt_config.h file from the connection sharing demo due to configuration logic in the demo's CMakeLists.txt which I have removed as part of this PR. 
Removing the include path config logic from connection sharing demo's CMakeLists.txt exposed that the shadow demo lacked the mqtt config file

Reply:
Ok. Thanks. That was a good find and a fix.